### PR TITLE
refactor: Make `SignedBlock` immutable

### LINKED ISSUE #4224: `SignedBlock` is immutable
After #4198 was merged the same should be done for `SignedBlock`

### PR DESCRIPTION
**Pull request by dima74**
## Description

As I understand, the problem is that `payload` of `SignedBlock` can be modified, which will result in invalid `SignedBlock` (inconsistent `payload` and `signatures`). Currently `payload` is modified only in tests. I made `payload` private and changed such tests to create new block instead of modifying existing one.

### Linked issue

Closes #4224

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: data_model/src/block.rs [hunk 1]
```diff
@@ -133,6 +133,20 @@ impl BlockHeader {
...
+    pub fn new(payload: BlockPayload, key_pair: &KeyPair) -> SignedBlockV1 {
+        let signatures = SignaturesOf::from(SignatureOf::new(key_pair, &payload));
+        SignedBlockV1 {
+            payload,
+            signatures,
+        }
+    }
+
+    /// Block payload. Used for tests
+    pub fn payload(&self) -> &BlockPayload {
```
Comment (mversic) [inline data_model/src/block.rs:146]: remove or make it `transparent_api`

File: data_model/src/block.rs [hunk 2]
```diff
@@ -133,6 +133,20 @@ impl BlockHeader {
 }
 
 impl SignedBlockV1 {
+    /// Create new signed block, using `key_pair` to sign `payload`
+    pub fn new(payload: BlockPayload, key_pair: &KeyPair) -> SignedBlockV1 {
```
Comment (mversic) [inline data_model/src/block.rs:137]:
```suggestion
    #[feature = "transparent_api"]
    pub fn new(payload: BlockPayload, key_pair: &KeyPair) -> SignedBlockV1 {
```

File: data_model/src/block.rs (hunk 1 above)
Comment (mversic) [inline data_model/src/block.rs:146]:
```suggestion
    #[feature = "transparent_api"]
    pub fn payload(&self) -> &BlockPayload {
```

**Review by Erigara — APPROVED** [review]

**Review by mversic — DISMISSED** [review]

**Comment by mversic** [comment]
your commit is not signed

**Comment by dima74** [comment]
> your commit is not signed

Fixed

**Review by mversic — APPROVED** [review]

**Review by Erigara — APPROVED** [review]

**Pull request merged**